frame/staking/src/pallet/impls.rs

```diff
@@ -292,23 +292,42 @@ impl<T: Config> Pallet<T> {
 	/// Actually make a payment to a staker. This uses the currency's reward function
 	/// to pay the right payee for the given staker account.
 	fn make_payout(stash: &T::AccountId, amount: BalanceOf<T>) -> Option<PositiveImbalanceOf<T>> {
-		let dest = Self::payee(stash);
-		match dest {
-			RewardDestination::Controller => Self::bonded(stash)
-				.map(|controller| T::Currency::deposit_creating(&controller, amount)),
-			RewardDestination::Stash => T::Currency::deposit_into_existing(stash, amount).ok(),
-			RewardDestination::Staked => Self::bonded(stash)
-				.and_then(|c| Self::ledger(&c).map(|l| (c, l)))
-				.and_then(|(controller, mut l)| {
-					l.active += amount;
-					l.total += amount;
-					let r = T::Currency::deposit_into_existing(stash, amount).ok();
+		// NOTE: temporary getter while `Payee` -> `Payees` lazy migration is taking place.
+		// Can replace with `dest = Self:payees(stash)` once migration is done.
+		let dest =
+			Self::get_payout_destination(stash, Self::bonded(stash).expect("bonded stash, qed."));
```

Comment:
Should not do `expect` here. This is assuming the caller function has ensured there is a bonded stash?
Anyway, we can noop if Self::bonded returns `None` like before.

Reply:
Yes it is assuming that the caller has already ensured bonded exists.